pkg/controller/shared_state.go

```diff
@@ -1531,14 +1530,16 @@ func (c *SharedState) createAzVolume(ctx context.Context, desiredAzVolume *azdis
 
 		azVolume.Status.Annotations = azureutils.AddToMap(azVolume.Status.Annotations, statusAnnotation...)
 		updated = azVolume.DeepCopy()
+		statusUpdateRequired = true
 
-	} else {
-		return nil
 	}
 
-	if _, err = c.azClient.DiskV1beta2().AzVolumes(c.config.ObjectNamespace).UpdateStatus(ctx, updated, metav1.UpdateOptions{}); err != nil {
-		return err
+	if statusUpdateRequired {
+		if _, err = c.azClient.DiskV1beta2().AzVolumes(c.config.ObjectNamespace).UpdateStatus(ctx, updated, metav1.UpdateOptions{}); err != nil {
```

Comment:
Shouldn't we use `UpdateCRIWithRetry` here? If we did, we wouldn't need `statusUpdateRequired` since it would no-op if there was no change.

Reply:
Good point.